### PR TITLE
Migrate from `purty` to `purs-tidy`

### PR DESCRIPTION
**Pull request by thomashoneyman**
👋 Hi @cprussin! This is a small pull request that migrates from `purty` to [`purs-tidy`](https://github.com/natefaubion/purescript-tidy) for formatting. I was working on the library a little and found that `purty` causes a few odd formatting decisions that aren't usually present in PureScript code, and we've seen a great result using `purs-tidy` at work at Awake, Halogen, the PureScript registry, and more.

In this PR I've kept the formatting changes minimal and preserved the arrow-last style used by `purty`. However, `purs-tidy` supports both arrow-first and arrow-last formatting, and so if you'd like to switch then I can push a commit changing that so you can see how it feels.

The biggest changes are in operators and in equational-style function declarations; I'll add comments to those directly to call them out in the review.

Of course, if you are not interested in switching then feel free to close this PR. Thanks!

### CODE REVIEW

File: docs/Examples/Middleware/Main.purs [hunk 1]
```diff
@@ -37,7 +37,6 @@ pathMiddleware ::
   HTTPure.Request ->
   HTTPure.ResponseM
 pathMiddleware _ { path: [ "middleware" ] } = HTTPure.ok "Middleware!"
-
 pathMiddleware router request = router request
```
Comment (thomashoneyman) [inline docs/Examples/Middleware/Main.purs:40]: `purs-tidy` doesn't insert newlines in equational-style declarations, which I think is the preferred way to write them so there's no confusion about whether it's a new declaration or not.

File: test/Test/HTTPure/TestHelpers.purs [hunk 2]
```diff
@@ -76,13 +70,12 @@ toBuffer response =
       stream = HTTPClient.responseAsStream response
     chunks <- Ref.new List.Nil
     Stream.onData stream $ \new -> Ref.modify_ (List.Cons new) chunks
-    Stream.onEnd stream
-      $ Ref.read chunks
+    Stream.onEnd stream $ Ref.read chunks
       >>= List.reverse
-      >>> Array.fromFoldable
-      >>> Buffer.concat
+        >>> Array.fromFoldable
```
Comment (thomashoneyman) [inline test/Test/HTTPure/TestHelpers.purs:75]: `purs-tidy` is aware of operator precedence, at least for operators defined in core and contrib libraries; if you would like the operator precedence to be generated for all code the library uses then you can generate an operator table using the `purs-tidy` CLI. I haven't done that here as it seems to be working well!

File: docs/Examples/Middleware/Main.purs (hunk 1 above)
Comment (cprussin) [reply docs/Examples/Middleware/Main.purs:40]: yeah, I never really liked these newlines either

File: test/Test/HTTPure/TestHelpers.purs (hunk 2 above)
Comment (cprussin) [reply test/Test/HTTPure/TestHelpers.purs:75]: Very nice, good to know, looks great as is though

**Review by cprussin — APPROVED** [review]
Beautiful, thanks @thomashoneyman

**Pull request merged**